src/Cli/Microsoft.DotNet.Cli.Utils/UILanguageOverride.cs

```diff
@@ -97,5 +99,31 @@ private static void SetIfNotAlreadySet(string environmentVariableName, int value
         {
             SetIfNotAlreadySet(environmentVariableName, value.ToString());
         }
+
+        private static bool CurrentWindowsVersionOfficiallySupportsUTF8Encoding()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Environment.OSVersion.Version.Major >= 10) // UTF-8 is only officially supported on 10+.
```

Comment:
Line 30 already checked that the platform is Windows. Maybe make this a more generic function `PlatformSupportsUtf8Encoding`. For now it only checks Windows, but will simplify the logic in the caller a bit.

Reply:
Thanks, that makes sense and maybe what @v-wuzhai was getting at (sorry.)

I changed it to `CurrentPlatformIsWindowsAndOfficiallySupportsUTF8Encoding` because we don't need to change the encoding on non windows platforms, and that is the reason the check was initially separated into is it windows, and then does this version of windows support the encoding, instead of a function that says if the platform does